service/src/main/java/io/camunda/service/AdHocSubprocessActivityServices.java

```diff
@@ -99,6 +112,50 @@ public List<AdHocSubprocessActivity> findActivatableActivities(
         "Failed to find Ad-Hoc Subprocess with ID '%s'".formatted(adHocSubprocessId));
   }
 
+  public CompletableFuture<ProcessInstanceModificationRecord> activateActivities(
+      final Long adHocSubprocessInstanceKey, final List<String> flowNodeIds) {
+    final var flowNode = flowNodeInstanceServices.getByKey(adHocSubprocessInstanceKey);
+
+    // TODO type returns as UNKNOWN - something to fix in exporter?
+    /*
+    if (flowNode.type() != FlowNodeType.AD_HOC_SUB_PROCESS) {
```

Comment:
This is `UNKNOWN` also on the ElasticSearch level (tested with `CamundaExporter` - maybe something we need to address separately? It would work also without this validation as we're anyways validating the list of activity IDs, but the returned error might be misleading as it states that the activities were not found (vs. the actual subprocess is missing/not a subprocess).